distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -691,16 +682,13 @@ type Vector a
              [1] + [2]
     + : Vector Any -> Vector Any
     + self that = case that of
-        _ : Vector ->
-            self_len = self.length
-            that_len = that.length
-            arr = Array.new (self_len + that_len)
-            Array.copy self.to_array 0 arr 0 self_len
-            Array.copy that.to_array 0 arr self_len that_len
-            Vector.from_polyglot_array arr
+        _ : Vector -> self.insert_builtin self.length that
         _ : Array -> self + Vector.from_polyglot_array that
         _ -> Error.throw (Type_Error.Error Vector that "that")
 
+    ## PRIVATE
+    insert_builtin self at values = @Builtin_Method("Vector.insert_builtin")
```

Comment:
If we made `insert_builtin` static method (here or elsewhere) we wouldn't need to define it in `Array`.